### ISSUE TITLE
Future enhancement to return element coordinates

### ISSUE THREAD

**Issue by ritchieanesco** [ZeeCoder/use-resize-observer#107]
Thank you for the great library!!

Just wondering if there were any plans to expose the element's `x`, `y`, `top`, `left`, etc. coordinates since you are already accessing dimensions from [contentRect](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserverEntry/contentRect)?

**Comment by ZeeCoder** [comment]
You're welcome! :pray: 
Hmm, maybe.

Currently onResize reports width and height, and it reports the requested kind of dimensions (content- or border box size), using borderBoxSize, contentBoxSize, or the old contentRect prop.

Only contentRect has properties other than width / height, and it seems a bit accidental if I'm honest from an early implementation, as outlined in the MDN docs.

I would rather consider a lower-level implementation to hook onto, something like `onEntry` reporting the entire entry object itself as an escape hatch, allowing to implement something like this. :thinking: 

So basically I would keep the current useResizeObserver as-is, to preserve its size observing API and optimisations, and could just add another lower-level hook called something like `useResizeObserverEntry`, with an onChange method :thinking:

**Comment by ZeeCoder** [comment]
ResizeObserverEntry is now exposed as one of the onResize callback's parameters, as `entry`, in v10 alpha and above now.
(use-resize-observer@alpha at the time of writing, just before releasing stable v10.)

So using [getBoundingClientRect](https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect) is now easier:
```js
const { ref } = useResizeObserver({
  onResize: ({ entry }) => {
    requestAnimationFrame(() => {
      const rect = entry.target.getBoundingClientRect();
      // ... do something with the result ...
    });
  }
});
```

In this example I used `requestAnimationFrame` to avoid [layout trashing](https://github.com/ZeeCoder/use-resize-observer/discussions/102#discussioncomment-5020345).
Only do this if you really need the x,y,top,left coordinates.

If you just need width/height, then just use the hook as normal, without accessing this new entry property.
